tests/crypto/src/common_test.c

```diff
@@ -127,6 +127,16 @@ uint32_t get_vector_count(const test_case_t *tc)
 	       test_vector_sizes[tc->vector_type];
 }
 
+size_t hex2bin_safe(const char *hex, uint8_t *buf, size_t buflen)
+{
+	if (hex == NULL) {
+		return 0;
+	}
+	else {
+		return hex2bin(hex, strlen(hex), buf, buflen);
+	}
```

Comment:
Could be a bit more terse using the ternary if.

```suggestion
        return hex == NULL ? 0 : hex2bin(hex, strlen(hex), buf, buflen);
```

Reply:
Yes, good point.